### PR TITLE
Refactor deposit

### PR DESCRIPTION
**Pull request by nkrivous**

### CODE REVIEW

File: packages/shared/components/deposit/components/deposit-popup.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+} from "gv-api-web";
+import React, { Fragment } from "react";
+import { connect } from "react-redux";
+import { Dispatch, bindActionCreators } from "redux";
+import { fetchBaseWallets } from "shared/components/wallet/services/wallet.services";
+import { ASSET, ROLE } from "shared/constants/constants";
+import RootState from "shared/reducers/root-reducer";
+import { RootThunk } from "shared/utils/types";
+
+import { DialogLoader } from "../../dialog/dialog-loader/dialog-loader";
```
Comment (pavelzubov) [inline packages/shared/components/deposit/components/deposit-popup.tsx]: Change to "shared/..."

File: packages/shared/components/deposit/components/deposit-top.tsx [hunk 2]
```diff
@@ -22,23 +16,32 @@ const DepositTop: React.FC<IDepositTop & InjectedTranslateProps> = ({
...
-        )}
-      </div>
+        </div>
+      )}
     </div>
   );
 };
 
-export default translate()(DepositTop);
+const DepositTop = translate()(_DepositTop);
```
Comment (pavelzubov) [inline packages/shared/components/deposit/components/deposit-top.tsx]: Add React.memo

File: packages/shared/components/deposit/components/program-deposit-container.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import RootState from "shared/reducers/root-reducer";
+import { ResponseError } from "shared/utils/types";
+
+import {
+  getProgramInfoCreator,
+  programInvestCreator
+} from "../services/program-deposit.service";
+import DepositPopup from "./deposit-popup";
+
+class _ProgramDepositContainer extends React.Component<Props, State> {
```
Comment (pavelzubov) [inline packages/shared/components/deposit/components/program-deposit-container.tsx]: Change to PureComponent

File: packages/shared/components/deposit/components/fund-deposit-container.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import RootState from "shared/reducers/root-reducer";
+import { ResponseError } from "shared/utils/types";
+
+import {
+  fundInvestCreator,
+  getFundInfoCreator
+} from "../services/fund-deposit.service";
+import DepositPopup from "./deposit-popup";
+
+class _FundDepositContainer extends React.Component<Props, State> {
```
Comment (pavelzubov) [inline packages/shared/components/deposit/components/fund-deposit-container.tsx]: Change to PureComponent

File: packages/investor/src/modules/fund-deposit/fund-deposit.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      onApply={onApply}
+      open={open}
+      onClose={onClose}
+      fundInvest={investorApi.v10InvestorFundsByIdInvestByAmountPost}
+      fetchInfo={investorApi.v10InvestorFundsByIdInvestInfoByCurrencyGet}
+    />
+  );
+};
+
+export default FundDeposit;
```
Comment (pavelzubov) [inline packages/investor/src/modules/fund-deposit/fund-deposit.tsx]: Add React.memo

File: packages/investor/src/modules/program-deposit/program-deposit.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      onApply={onApply}
+      open={open}
+      onClose={onClose}
+      programInvest={investorApi.v10InvestorProgramsByIdInvestByAmountPost}
+      fetchInfo={investorApi.v10InvestorProgramsByIdInvestInfoByCurrencyGet}
+    />
+  );
+};
+
+export default ProgramDeposit;
```
Comment (pavelzubov) [inline packages/investor/src/modules/program-deposit/program-deposit.tsx]: Add React.memo

File: packages/manager/src/modules/fund-deposit/fund-deposit.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      onApply={onApply}
+      open={open}
+      onClose={onClose}
+      fundInvest={managerApi.v10ManagerFundsByIdInvestByAmountPost}
+      fetchInfo={managerApi.v10ManagerFundsByIdInvestInfoByCurrencyGet}
+    />
+  );
+};
+
+export default FundDeposit;
```
Comment (pavelzubov) [inline packages/manager/src/modules/fund-deposit/fund-deposit.tsx]: Add React.memo

File: packages/manager/src/modules/program-deposit/program-deposit.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      onApply={onApply}
+      open={open}
+      onClose={onClose}
+      programInvest={managerApi.v10ManagerProgramsByIdInvestByAmountPost}
+      fetchInfo={managerApi.v10ManagerProgramsByIdInvestInfoByCurrencyGet}
+    />
+  );
+};
+
+export default ProgramDeposit;
```
Comment (pavelzubov) [inline packages/manager/src/modules/program-deposit/program-deposit.tsx]: Add React.memo

**Comment by nkrivous** [comment]
Done

**Pull request merged**